data/api/client-server/appservice_ping.yaml

```
@@ -0,0 +1,148 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+swagger: '2.0'
+info:
+  title: "Matrix Client-Server Application Service Ping API"
+  version: "1.0.0"
+host: localhost:8008
+schemes:
+  - https
+  - http
+basePath: /_matrix/client/v1
+consumes:
+  - application/json
+produces:
+  - application/json
+securityDefinitions:
+  # Note: this is the same access_token definition used elsewhere in the client
+  # server API, however this expects an access token for an application service.
+  $ref: definitions/security.yaml
+paths:
+  "/appservice/{appserviceId}/ping":
+    post:
+      x-addedInMatrixVersion: "1.7"
+      summary: |-
+        Ask the homeserver to ping the application service to ensure the connection works.
+      description: |-
+        This API asks the homeserver to call the `/_matrix/app/v1/ping` endpoint
+        on the application service to ensure that the homeserver can communicate
+        with the application service.
+
+        This API requires the use of an application service access token (`as_token`)
+        instead of a typical client's access_token. This API cannot be invoked by
+        users who are not identified as application services. Additionally, the
+        appservice ID in the path must be the same as the appservice whose `as_token`
+        is being used.
+      operationId: pingAppservice
+      parameters:
+        - in: path
+          type: string
+          name: appserviceId
+          description: |-
+            The appservice ID of the appservice to ping. This must be the same
+            as the appservice whose `as_token` is being used to authenticate the
+            request.
+          required: true
+          x-example: "telegram"
+        - in: body
+          name: body
+          required: true
+          schema:
+            type: object
+            properties:
+              transaction_id:
+                type: string
+                description: |-
+                  An optional transaction ID that is passed through to the `/_matrix/app/v1/ping` call.
+                example: "mautrix-go_1683636478256400935_123"
+      security:
+        # again, this is the appservice's token - not a typical client's
+        - accessToken: []
+      responses:
+        200:
+          description: The ping was successful
+          schema:
+            type: object
+            properties:
+              duration_ms:
+                type: integer
+                description: |-
+                    The duration that the `/_matrix/app/v1/ping` request took
+                    from the homeserver's point of view.
```

It's obvious from the name, but best to be explicit. Also let's link the endpoint:

```suggestion
                    The duration in milliseconds that the
                    [`/_matrix/app/v1/ping`](#post_matrixappv1ping)
                    request took from the homeserver's point of view.
```